go.mod

```diff
@@ -803,3 +809,14 @@ exclude (
 replace go.opentelemetry.io/collector => github.com/grafana/opentelemetry-collector v0.4.1-0.20240429170914-d1e101852ba5
 
 replace github.com/prometheus/procfs => github.com/prometheus/procfs v0.12.0
+
+// Replaces required for Beyla to implement metric expiration
+// Remove once this PR is merged upstream:
+// https://github.com/open-telemetry/opentelemetry-specification/pull/4135
+replace (
+	go.opentelemetry.io/otel => github.com/grafana/opentelemetry-go v1.28.0-grafana.2
+	go.opentelemetry.io/otel/metric => github.com/grafana/opentelemetry-go/metric v1.28.0-grafana.2
+	go.opentelemetry.io/otel/sdk => github.com/grafana/opentelemetry-go/sdk v1.28.0-grafana.2
+	go.opentelemetry.io/otel/sdk/metric => github.com/grafana/opentelemetry-go/sdk/metric v1.28.0-grafana.2
+	go.opentelemetry.io/otel/trace => github.com/grafana/opentelemetry-go/trace v1.28.0-grafana.2
+)
```

Comment:
It would be much better if upstream's PR would be merged before to avoid the replace directives. Updating the Otel dependencies in Alloy is already a long and difficult process (see the current PR to update it: https://github.com/grafana/alloy/pull/1362). We update our Otel dependencies before every release (every 6 weeks). Having the fork means that we will also need to keep the fork up to date every 6 weeks until the PR is merged. What if it's not merged or it takes a different direction?